### PR TITLE
Verification for explicit colored petri nets

### PR DESCRIPTION
**Pull request by JE-FH**
review and merge #5  first

### CODE REVIEW

File: src/PetriEngine/ExplicitColored/CPNMultiSet.cpp [hunk 1]
```diff
@@ -2,109 +2,51 @@
...
-                        continue;
-                    }
-
-                    if (b.second > a.second) {
-                        _cardinality -= a.second;
-                        a.second = 0;
+            for (auto& otherCount : other._counts) {
+                auto it = _counts.find(otherCount.first);
+                if (it == _counts.end()) {
+                    _counts.emplace(otherCount);
```
Comment (emilnormann) [inline src/PetriEngine/ExplicitColored/CPNMultiSet.cpp:41]: Why

File: src/PetriEngine/ExplicitColored/CPNMultiSet.cpp [hunk 2]
```diff
@@ -2,109 +2,51 @@
...
+                } else {
+                    if (otherCount.second > it->second) {
+                        _cardinality -= it->second;
+                        it->second = 0;
                     } else {
-                        _cardinality = (_cardinality + b.second) - a.second;
-                        a.second -= b.second;
+                        _cardinality -= it->second;
+                        it->second -= it->second;
```
Comment (emilnormann) [inline src/PetriEngine/ExplicitColored/CPNMultiSet.cpp:48]: _cardinality -= otherCount.second
it->second -= otherCount.second
right?

**Review by emilnormann — CHANGES_REQUESTED** [review]
It is probably mostly lgtm

File: src/PetriEngine/ExplicitColored/CPNMultiSet.cpp (hunk 2 above)
Comment (JE-FH) [reply src/PetriEngine/ExplicitColored/CPNMultiSet.cpp]: True and based

File: src/PetriEngine/ExplicitColored/CPNMultiSet.cpp (hunk 1 above)
Comment (JE-FH) [reply src/PetriEngine/ExplicitColored/CPNMultiSet.cpp]: No good reason

**Pull request merged**